addons/stock_picking_batch/models/stock_move_line.py

```diff
@@ -55,13 +55,14 @@ def _add_to_wave(self, wave=False):
                 qty = line.product_uom_id._compute_quantity(line.quantity, line.product_id.uom_id, rounding_method='HALF-UP')
                 qty_by_move[line.move_id] += qty
 
+            # If all moves are to be transferred to the wave, link the picking to the wave
             if lines == picking.move_line_ids and lines.move_id == picking.move_ids:
-                move_complete = True
+                add_all_moves = True
                 for move, qty in qty_by_move.items():
-                    if float_compare(move.product_qty, qty, precision_rounding=move.product_uom.rounding) != 0:
-                        move_complete = False
+                    if float_is_zero(qty, precision_rounding=move.product_uom.rounding) != 0:
```

Comment:
detail but since the method return True or False `!= 0` is confusing me, you could just do:

```suggestion
                    if float_is_zero(qty, precision_rounding=move.product_uom.rounding):
```